cmd/aws-iam-authenticator/root.go

```diff
@@ -102,6 +102,7 @@ func getConfig() (config.Config, error) {
 		EC2DescribeInstancesQps:           viper.GetInt("server.ec2DescribeInstancesQps"),
 		EC2DescribeInstancesBurst:         viper.GetInt("server.ec2DescribeInstancesBurst"),
 		ScrubbedAWSAccounts:               viper.GetStringSlice("server.scrubbedAccounts"),
+		DynamicFilePath:                   viper.GetString("server.dynamicfilepath"),
 	}
 	if err := viper.UnmarshalKey("server.mapRoles", &cfg.RoleMappings); err != nil {
```

Comment:
I think we should define an "NestedMappings" struct in `[pkg/config/types.go](https://github.com/kubernetes-sigs/aws-iam-authenticator/blob/master/pkg/config/types.go)` and embed it in both the viper config as well as use it for the dynamic file mode. There will be minimal changes required for the existing code and a lot of cutting of the code from this CR.

I made an example of how it works here: https://go.dev/play/p/RXYXdE3mvoI

Additional resources:
* https://gobyexample.com/struct-embedding
* https://go.dev/doc/effective_go#embedding